scene/main/canvas_item.cpp

```diff
@@ -1511,6 +1511,7 @@ void CanvasTexture::set_normal_texture(const Ref<Texture2D> &p_normal) {
 	normal_texture = p_normal;
 	RID tex_rid = normal_texture.is_valid() ? normal_texture->get_rid() : RID();
 	RS::get_singleton()->canvas_texture_set_channel(canvas_texture, RS::CANVAS_TEXTURE_CHANNEL_NORMAL, tex_rid);
+	emit_changed();
```

Comment:
I feel like all these setters should have early returns if you attempt to assign the same value twice. Especially with the signal added now.